### PR TITLE
KAFKA-6476: Documentation for dynamic broker configuration

### PR DESCRIPTION
**Pull request by rajinisivaram**
Docs for dynamic broker configuration (KIP-226).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@hachikuji Can you review, please? Thank you!

File: docs/security.html [hunk 1]
```diff
@@ -246,6 +248,35 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
...
+        listener.name.sasl_ssl.plain.sasl.jaas.config= org.apache.kafka.common.security.plain.PlainLoginModule required \
+            username="admin" \
+            password="admin-secret" \
+            user_admin="admin-secret" \
+            user_alice="alice-secret";</pre>
+
+            If JAAS configuration is defined at different levels, the order of precedence used is:
+            <ul>
+              <li><code>listener.name.{listenerName}.{saslMechanism}.sasl.jaas.config</code></li>
+              <li><code>sasl.jaas.config</code></li>
```
Comment (omkreddy) [inline docs/security.html]: Above we mentioned "The property name must be prefixed with the listener prefix including the SASL mechanism".  If we give "sasl.jaas.config" alone, what is the behaviour?

I am getting error when I start broker with "sasl.jaas.config" config property .

Comment (rajinisivaram) [reply docs/security.html]: @omkreddy Thanks for the review. `sasl.jaas.config` should be prefixed with listener name and mechanism for brokers. It shouldn't have been included here, updated.

File: docs/configuration.html [hunk 2]
```diff
@@ -33,6 +33,161 @@ <h3><a id="brokerconfigs" href="#brokerconfigs">3.1 Broker Configs</a></h3>
...
+  <li>Dynamic cluster-wide default config stored in ZooKeeper</li>
+  <li>Static broker config from <code>server.properties</code></li>
+  <li>Kafka default, see <a href="#brokerconfigs">broker configs</a></li>
+  </ul>
+
+  <h5>Updating Password Configs Dynamically</h5>
+  <p>Password config values that are dynamically updated are encrypted before storing in ZooKeeper. The broker config
+  <code>password.encoder.secret</code> must be configured in <code>server.properties</code> to enable dynamic update
+  of password configs. The secret may be different on different brokers.</p>
+  <p>The secret used for password encoding may be rotated with rolling restart of brokers. The old secret used for encoding
```
Comment (hachikuji) [inline docs/configuration.html]: nit: with _a_ rolling restart?

File: docs/configuration.html [hunk 3]
```diff
@@ -33,6 +33,161 @@ <h3><a id="brokerconfigs" href="#brokerconfigs">3.1 Broker Configs</a></h3>
...
+  passwords currently in ZooKeeper must be provided in the static broker config <code>password.encoder.old.secret</code> and
+  the new secret must be provided in <code>password.encoder.secret</code>. All dynamic password configs stored in ZooKeeper
+  will be re-encoded with the new secret when the broker starts up.</p>
+  <p>In Kafka 1.1.0, all dynamically updated password configs must be provided in every alter request when updating configs
```
Comment (hachikuji) [inline docs/configuration.html]: A few of these comments mention 1.1.0, but I don't think any of the suggested improvements would be fixed in 1.1.1 or another bug fix release. Could we make the comments more general so we don't need to remember to update them?

File: docs/configuration.html [hunk 4]
```diff
@@ -33,6 +33,161 @@ <h3><a id="brokerconfigs" href="#brokerconfigs">3.1 Broker Configs</a></h3>
...
+  with the listener and mechanism prefix. See <a href="#security_jaas_broker">JAAS configuration for Kafka brokers</a> for details.</p>
+
+  <p>In Kafka version 1.1.0, the listener used by the inter-broker listener may not be updated dynamically. To update the inter-broker
+  listener to a new listener, the new listener may be added on all brokers without restarting the broker. A rolling restart is then
+  required to update <code>inter.broker.listener.name</code>.</p>
+
+  In addition to all the security configs of new listeners, the following configs may be updated dynamically at per-broker level:
+  <ul>
+    <li><code>listeners</code></li>
+    <li><code>advertised.listeners</code></li>
```
Comment (hachikuji) [inline docs/configuration.html:186]: Is it worth mentioning that the inter-broker listener must still be included?

**Review by hachikuji — APPROVED** [review]
LGTM. Very thorough and well-written! Left a couple trivial comments which do not require re-review.

**Comment by rajinisivaram** [comment]
@hachikuji Thanks for the review, addressed the comments, merging to trunk and 1.1.

**Pull request merged**